app/streams/publishing_api/message_handler.rb

```diff
@@ -7,39 +7,61 @@ def self.process(*args)
 
   def initialize(message)
     @message = message
-
-    @new_item = PublishingAPI::MessageAdapter.to_dimension_item(message)
-    @old_item = Dimensions::Item.find_by(base_path: new_item.base_path, latest: true)
   end
 
   def process
-    return unless new_item.older_than?(old_item)
-
-    if is_links_update?
-      grow_dimension! if links_have_changed?
-    else
-      grow_dimension!
+    items.each do |item|
+      if is_links_update?
+        item.process_links!
+      else
+        item.process!
+      end
     end
   end
 
 private
 
-  attr_reader :message, :new_item, :old_item
+  attr_reader :message
 
-  def is_links_update?
-    routing_key = message.delivery_info.routing_key
-    routing_key.ends_with?('.links')
+  def get_old_items(new_base_paths)
+    content_id = message.payload["content_id"]
+
+    # Get everything with the same content id, so we can deprecate parts
```

Comment:
I would extract this query into the `Item` and I would use the tests to describe the how it behaves. This way we can get rid of the comment as well.